clients/java/signalr/src/main/java/ConsoleLogger.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+public class ConsoleLogger implements Logger {
+    private LogLevel logLevel;
+    public ConsoleLogger(LogLevel logLevel) {
+            this.logLevel = logLevel;
+    }
+
+    @Override
+    public void log(LogLevel logLevel, String message) {
+        if(logLevel.value >= this.logLevel.value){
```

Comment:
Any custom logger will have to have this logic which is a smell (coupling). An idea I came up with in the Swift client was to create an internal filtering logger that wraps the actual logger (https://github.com/moozzyk/SignalR-Client-Swift/commit/afc00220754babf78054a8f66abc82aa1ef08e1d). This way none of the implementations of the logger interface needs to worry about min log level.